types/share_splitting.go

```diff
@@ -47,8 +62,10 @@ func (msw *MessageShareWriter) Export() NamespacedShares {
 	return shares
 }
 
+// note: as an optimization we can probably get rid of this if we just add
+// checks each time we write.
 func (msw *MessageShareWriter) sortMsgs() {
-	sort.Slice(msw.shares, func(i, j int) bool {
+	sort.SliceStable(msw.shares, func(i, j int) bool {
```

Comment:
this probably could have been its own PR, as this is desirable even without non-interactive defaults, but it is specifically needed for padding so I thought I'd include it here